src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+import { error } from 'console';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public shape: 'triangle',
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (!['red', 'blue', 'green'].includes(color)) {
+      throw new Error('wrong color');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('the side of the triangle is less than or equal to zero');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('the side of the triangle is less than or equal to zero');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  constructor(
+    public shape: 'circle',
+    public color: string,
+    public radius: number,
+  ) {
+    if (!['red', 'blue', 'green'].includes(color)) {
+      throw new Error('wrong color');
+    }
 
-export class Rectangle implements Figure {}
+    if (radius < 0) {
+      throw new Error('radius must be > than zero');
```

Comment:
[CODE STYLE] - The error message should be more descriptive. Consider specifying that the radius must be a positive number.